miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditLibraryController.java

```diff
@@ -845,6 +866,10 @@ public ModelAndView editPropagateSamples(@PathVariable String sampleIds, ModelMa
           hasPlain = true;
         }
         LibraryDto library = new LibraryDto();
+        Library tempLibrary = new LibraryImpl();
+        tempLibrary.setSample(sample);
+        final String libraryAlias = namingScheme.generateLibraryAlias(tempLibrary);
```

Comment:
That should be changed to match how the sample forms work: If sample alias generation is enabled, it is not a required field. It may still be provided if you'd like to use a custom alias, but leaving it blank will result in one being generated.

Generating the alias during save is also better for generating unique names. If it's done during creation and two people are creating libraries at the same time, it can result in a collision.